plugins/database/mysql/mysql.go

```diff
@@ -0,0 +1,187 @@
+package mysql
+
+import (
+	"database/sql"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/vault/builtin/logical/database/dbplugin"
+	"github.com/hashicorp/vault/helper/strutil"
+	"github.com/hashicorp/vault/plugins/helper/database/connutil"
+	"github.com/hashicorp/vault/plugins/helper/database/credsutil"
+	"github.com/hashicorp/vault/plugins/helper/database/dbutil"
+)
+
+const defaultMysqlRevocationStmts = `
+	REVOKE ALL PRIVILEGES, GRANT OPTION FROM '{{name}}'@'%'; 
+	DROP USER '{{name}}'@'%'
+`
+const mySQLTypeName = "mysql"
+
+type MySQL struct {
+	connutil.ConnectionProducer
+	credsutil.CredentialsProducer
+}
+
+// New implements builtinplugins.BuiltinFactory
+func New() (interface{}, error) {
+	connProducer := &connutil.SQLConnectionProducer{}
+	connProducer.Type = mySQLTypeName
+
+	credsProducer := &credsutil.SQLCredentialsProducer{
+		DisplayNameLen: 4,
+		UsernameLen:    16,
```

Comment:
Because this value changed in different versions of mysql/maria I think of the ability to configure as being separate from whether it should be a different plugin, so that you don't need e.g. mysql-32 and mysql-16 plugins.